frontend/src/components/file-explorer/FileExplorer.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import {
+  IoIosArrowBack,
+  IoIosArrowForward,
+  IoIosRefresh,
+} from "react-icons/io";
+import { twMerge } from "tailwind-merge";
+import { getWorkspace } from "../../services/fileService";
+import ExplorerTree from "./ExplorerTree";
+import { removeEmptyNodes } from "./utils";
+import IconButton from "../IconButton";
+
+interface ExplorerActionsProps {
+  onRefresh: () => void;
+  toggleHidden: () => void;
+  isHidden: boolean;
+}
+
+function ExplorerActions({
+  toggleHidden,
+  onRefresh,
+  isHidden,
+}: ExplorerActionsProps) {
+  return (
+    <div
+      className={twMerge(
+        "transform flex h-[24px] items-center gap-1 absolute top-4 right-2",
+        isHidden ? "right-3" : "right-2",
+      )}
+    >
+      {!isHidden && (
+        <IconButton
+          icon={
+            <IoIosRefresh
+              size={20}
+              className="text-neutral-400 hover:text-neutral-100 transition"
+            />
+          }
+          testId="refresh"
+          ariaLabel="Refresh workspace"
+          onClick={onRefresh}
+        />
+      )}
+
+      <IconButton
+        icon={
+          isHidden ? (
+            <IoIosArrowForward
+              size={20}
+              className="text-neutral-400 hover:text-neutral-100 transition"
+            />
+          ) : (
+            <IoIosArrowBack
+              size={20}
+              className="text-neutral-400 hover:text-neutral-100 transition"
+            />
+          )
+        }
+        testId="close"
+        ariaLabel="Close workspace"
+        onClick={toggleHidden}
+      />
+    </div>
+  );
+}
+
+interface FileExplorerProps {
+  onFileClick: (path: string) => void;
+}
+
+function FileExplorer({ onFileClick }: FileExplorerProps) {
+  const [workspace, setWorkspace] = React.useState<TreeNode[]>([]);
```

Comment:
If we're using the currently implemented state, this cannot be an array.

Reply:
You're right, this small change was part of the "multi-root" feature that came with the new setup. It is probably best to remove this